IronPythonExtension/IronPythonExtension.cs

```diff
@@ -61,26 +62,9 @@ public void Startup(StartupParams sp)
         /// <param name="sp"></param>
         public void Ready(ReadyParams sp)
         {
-            // Searches for DSIronPython engine binary in same folder with extension itself
-            var targetDir = Path.GetDirectoryName(Assembly.GetAssembly(typeof(IronPythonExtension)).Location);
-            var libraryLoader = sp.StartupParams.LibraryLoader;
-            Assembly pythonEvaluatorLib = null;
-            try
-            {
-                pythonEvaluatorLib = Assembly.LoadFrom(Path.Combine(targetDir, PythonEvaluatorAssembly + ".dll"));
-            }
-            catch (Exception ex)
-            {
-                // Most likely the IronPython engine is excluded in this case
-                // but logging the exception message in case for diagnose
-                OnMessageLogged(LogMessage.Info(ex.Message));
-                return;
-            }
-            // Import IronPython Engine into VM, so Python node using IronPython engine could evaluate correctly
-            if (pythonEvaluatorLib != null)
-            {
-                libraryLoader.LoadNodeLibrary(pythonEvaluatorLib);
-            }
+            var extraPath = Path.Combine(new FileInfo(Assembly.GetAssembly(typeof(IronPythonExtension)).Location).Directory.Parent.FullName, "extra");
+            var alc = new IsolatedPythoContext(Path.Combine(extraPath,"DSIronPython.dll"));
+            alc.LoadFromAssemblyName(new AssemblyName("DSIronPython"));
```

Comment:
this means that only the DSIronPython.dll  assembly (and its own dependencies) will be isolated 
The IronPythonExtension assembly will be part of Dynamo's ALC

This sounds fine to me. I did the same thing with Dynamo4Revit and it worked ok
As long as the IronPythonExtension 's dependencies are ok to be in Dynamo's ALC

Reply:
Also I do not remember how the dependencies of the entrypoint (DSIronPython.dll) are resolved.
Hopefully it happens magically by using the ALC resolver and the deps.json file

I remember there were some issues with the dependencies that required special Resolver code in the Host app (that would be Dynamo in this case)

Reply:
yes, correct `IronPythonExtension` basically has no dependencies except for Dynamo and at least my testing indicates that the assemblies are isolated from the default ALC (since ironPython3 (not isolated) works side by side))

I think I can double check by using The GetAssemblyLoadContext() method on each assembly, or maybe using dotnet-trace tooling see exactly when the load occurs.

Reply:
Harder to do, but you can simply put breakpoints in all the resolvers in Dynamo and in this extension in the Assembly Load(AssemblyName assemblyName) and implement https://learn.microsoft.com/en-us/dotnet/api/system.runtime.loader.assemblyloadcontext.resolving?view=net-8.0
You could implement that in the dynamo default ALC and put a breakpoint too

Reply:
Here is the algorithm, we use the name variant:
https://learn.microsoft.com/en-us/dotnet/core/dependency-loading/loading-managed#algorithm

I'm pretty sure that dependencies of the assembly that are loaded using by name - are loaded into the same ALC, or at least as indicated by the algorithm, an attempt is made to resolve them before falling back to the default ALC.

Image shows the dependencies loaded side by side 

![Screenshot 2023-12-11 at 5 19 45 PM](https://github.com/DynamoDS/DSIronPython/assets/508936/7b50a042-b46e-4a28-853f-fc834ddb9a33)

 
![Screenshot 2023-12-11 at 5 26 15 PM](https://github.com/DynamoDS/DSIronPython/assets/508936/72f205d8-cd72-4691-a127-edc9e719aa50)

I checked the above with the deps.json files in the DSIronPython folder, and without them and got the same functioning behavior of side by side iron python versions and the same binaries loaded into the ALC.

Reply:
This was the issue I was thinking of https://github.com/dotnet/runtime/issues/93780#issuecomment-1787339493
But this does not seem to be the case with IronPython deps. They probably are all in the resolver folder